planner/core/logical_plan_builder.go

```diff
@@ -312,6 +312,12 @@ func (p *LogicalJoin) extractOnCondition(conditions []expression.Expression, der
 
 func extractTableAlias(p LogicalPlan) *model.CIStr {
 	if p.Schema().Len() > 0 && p.Schema().Columns[0].TblName.L != "" {
+		tblName := p.Schema().Columns[0].TblName.L
+		for _, column := range p.Schema().Columns {
+			if column.TblName.L != tblName {
```

Comment:
We can just check the first column and the last column?

Reply:
I have considered this, but kenan suggested that it would be better not to do so, it's not a general solution 0.0
We are going to add more Hint for TiDB, and Schema is still being refactored, so I think this part may need to be developed in the future.